web/pages/settings/index.tsx

```diff
@@ -0,0 +1,286 @@
+import { db } from "@lib/firebase";
+import { doc, getDoc, updateDoc, query, collection, getDocs, orderBy } from "firebase/firestore";
+import { GetServerSideProps, GetStaticProps } from "next";
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import { getAuth, reauthenticateWithCredential, EmailAuthProvider, updatePassword } from "firebase/auth";
+import Waivers from "./waivers/index";
+import { formatDoc } from "@lib/firebase/getDoc";
+import { Waiver } from "@lib/types";
+
+type PhoneSettings = {
+  phoneNumber: string;
+};
+
+interface Props {
+  waivers: Waiver[];
+}
+
+const AccountTab = ({ phoneNumber }: PhoneSettings) => {
+  const [phNumber, setPhNumberState] = useState(phoneNumber);
+  const [editPhone, setEditPhone] = useState(false);
+  const [password, setPassword] = useState("");
+  const [editPassword, setEditPassword] = useState(false);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<PhoneSettings>();
+
+  const auth = getAuth();
+  const user = auth.currentUser;
+
+  const onPhoneNumberSubmit = handleSubmit(async (data) => {
+    const newNumber = data.phoneNumber;
+    if (newNumber != phNumber) {
+      const settingsRef = doc(db, "app", "settings");
+      await updateDoc(settingsRef, {
+        contact: {
+          phone: newNumber,
+        },
+      });
+      alert("Phone number updated");
+      setPhNumberState(newNumber);
+      setEditPhone(false);
+    }
+  });
+
+  const validateOldPassword = (oldPassword: string) => {
+    if (user && user.email) {
+      const credential = EmailAuthProvider.credential(
+        user.email,
+        oldPassword
+      );
+      return reauthenticateWithCredential(user, credential).then(()=>false).catch(()=>true);
+    }
+    return false;
+  }
+  
+  const onPasswordSubmit = handleSubmit(async () => {
+    if (user) {
+      updatePassword(user, password).then(() => {
+        alert("Password updated");
+      })
+    }
+  })
```

Comment:
```suggestion
   try {
      if (user) {
        await updatePassword(user, password)
        alert("Password updated")
      }
   } catch (e) {
    // Handle error...
   }  
  })
```
I suggest you handle error for all of these fields, but this one especially!